test/cases/compile_errors/non_error_sets_used_in_merge_error_sets_operator.zig

```diff
@@ -0,0 +1,15 @@
+export fn foo() void {
+    const Errors = u8 || u16;
+    _ = Errors;
+}
+export fn bar() void {
+    const Errors = error{} || u16;
+    _ = Errors;
+}
+
+// error
+// backend=stage2
+// target=native
+//
+// :2:20: error: expected error set type, found 'u8'
+// :6:31: error: expected error set type, found 'u16'
```

Comment:
We seem to have lost the handy note:
```
note: `||` merges error sets; `or` performs boolean OR
```
this is beneficial because `||` in other languages is `or`.

I haven't been checking too closely on your other commits, have you been making sure that useful error notes are preserved when marking these test cases as "passing" for stage2?

Reply:
This note is still there but only when both operands are bools.